lib/new_relic/agent/instrumentation/active_job_subscriber.rb

```diff
@@ -0,0 +1,55 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'new_relic/agent/instrumentation/notifications_subscriber'
+
+module NewRelic
+  module Agent
+    module Instrumentation
+      class ActiveJobSubscriber < NotificationsSubscriber
+        PAYLOAD_KEYS = %i[adapter db_runtime error job wait]
+
+        def start_segment(name, id, payload)
+          segment = Tracer.start_segment(name: metric_name(name, payload))
+
+          PAYLOAD_KEYS.each do |key|
+            segment.params[key] = payload[key] if payload.key?(key)
+          end
+
+          push_segment(id, segment)
+        end
+
+        def finish_segment(id, payload)
+          if segment = pop_segment(id)
+            if exception = exception_object(payload)
+              segment.notice_error(exception)
+            end
+            segment.finish
+          end
+        end
```

Comment:
This method looks very similar to the one in `NotificationsSubscriber`. Do we need it here too?

Reply:
The work on this PR began before `NotificationsSubscriber` delivered this method and I missed that. I knew about `start` and `finish`, but not `start_segment` and `finish_segment`. Will update.

Reply:
Addressed with [6e2aaf5](https://github.com/newrelic/newrelic-ruby-agent/pull/1761/commits/6e2aaf5f00b3447670f68fe508a2d7c52bd6263d)

Reply:
Thank you